posts/views.py

```diff
@@ -586,96 +588,142 @@ def post_preview_image(request: Request, pk):
 
 @api_view(["GET"])
 @permission_classes([AllowAny])
-def download_csv(request, pk: int):
+def download_data(request, pk: int):
```

Comment:
I'd moved all GET params definitions to the separate `serializers.Serializer` class to make it easier to read. You can still keep validation & querysets in this view though